client/src/utils/auth.ts

```diff
@@ -1,30 +1,48 @@
 import { createContext } from "react";
-import { auth, IUser } from "config/firebase";
+import { auth, IUser, db } from "config/firebase";
+import "firebase/firestore";
 
 // TODO: add other relevant user info to context type
 export interface AuthContext {
   user: IUser;
+  userInfo: UserInfo | null;
 }
 
 export const UserContext = createContext<AuthContext>({
   user: null,
+  userInfo: null,
 });
 
-export interface UserInfo {
+export interface UserAuthInfo {
   email: string;
   password: string;
 }
 
+export interface UserInfo {
+  email: string;
+  password?: string; //used only for auth, not passed in other queries
+  firstName: string;
+  lastName: string;
+  isAdmin: number;
+}
+
 /**
  * @desc Register user
  * @return
  * @param {*} user
  */
 export const registerUser = async (userinfo: UserInfo): Promise<any> => {
   return auth
-    .createUserWithEmailAndPassword(userinfo.email, userinfo.password)
+    .createUserWithEmailAndPassword(userinfo.email, userinfo.password!) //password will definitely be passed in registration
     .then((user) => {
-      // TODO: Add other user provided data to firestore
+      db.collection("users")
+        .doc(auth.currentUser!.uid)
```

Comment:
It might be cleaner/easier to just use the user object that's passed down here ( `user.uid` ).